File: tests/07_phases_ompss.dg/fortran/success_task_01.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>
PROGRAM P
    INTEGER :: X(20, 10)

    X = 1

    !$OMP TASK INOUT(X(2, 1))
       X(2, 1) = X(2, 1) + 41
    !$OMP END TASK 


    !$OMP TASK IN(X(2, 1))
        IF (X(2,1) /= 42) STOP 1
    !$OMP END TASK 

    !$OMP TASKWAIT
END PROGRAM P

File: tests/07_phases_ompss_2.dg/cxx/success_task_01.cpp
/*
<testinfo>
test_generator="config/mercurium-ompss-2"
test_nolink=yes
</testinfo>
*/

#pragma oss task
void f()
{ }

void g() {
    #pragma oss task
    {
        f();
    }

    #pragma oss loop chunksize(4)
    for (int i = 0; i < 100; ++i)
    {}

    #pragma oss taskwait
}
